packages/collector/test/nativeModuleRetry/test.js

```diff
@@ -91,114 +57,97 @@ describe('retry loading native addons', function () {
+          expect(fs.existsSync(path.join(directoryPath, 'src', 'eventLoopStats.cc'))).to.be.true;
+          expect(fs.existsSync(path.join(directoryPath, 'src', 'eventLoopStats.js'))).to.be.true;
+          expect(fs.existsSync(path.join(directoryPath, 'build', 'Release', 'eventLoopStats.node'))).to.be.true;
+        }
+
+        if (directoryPath.includes('gcstats.js')) {
+          expect(fs.existsSync(path.join(directoryPath, 'src', 'gcstats.cc'))).to.be.true;
+          expect(fs.existsSync(path.join(directoryPath, 'src', 'gcstats.js'))).to.be.true;
+          expect(fs.existsSync(path.join(directoryPath, 'build', 'Release', 'gcstats.node'))).to.be.true;
+        }
+      };
+
+      before(async () => {
+        execSync(`rm -rf ${tmpFolder}`, { cwd: __dirname, stdio: 'inherit' });
+        execSync(`mkdir -p ${tmpFolder}`, { cwd: __dirname, stdio: 'inherit' });
+
+        execSync(`cp app.js package.json ${tmpFolder}/`, { cwd: __dirname, stdio: 'inherit' });
+
+        // eslint-disable-next-line no-console
+        console.log('Running npm install in', tmpFolder);
+
+        execSync('rm -rf node_modules', { cwd: tmpFolder, stdio: 'inherit' });
+        execSync('npm install --prefix ./', { cwd: tmpFolder, stdio: 'inherit' });
+        execSync(`rm -rf node_modules/${opts.name}`, { cwd: tmpFolder, stdio: 'inherit' });
+
+        if (fs.existsSync(path.join(tmpFolder, 'node_modules', opts.name))) {
+          throw new Error('The module should not be present at this point');
+        }
+
+        // Currently we do not ship darwin prebuilds via shared-metrics
+        // See package.json
+        execSync(
+          `cp -R ../../../shared-metrics/addons/darwin ${tmpFolder}/node_modules/@instana/shared-metrics/addons/`,
+          {
+            cwd: __dirname,
+            stdio: 'inherit'
+          }
+        );
+
+        controls = new ProcessControls({
+          appPath: path.join(tmpFolder, 'app'),
+          agentControls,
+          useGlobalAgent: true,
+          env: {
+            DEV_PATH: __dirname
+          }
+        });
+
+        await controls.startAndWaitForAgentConnection();
+      });
+
+      beforeEach(async () => {
+        await agentControls.clearReceivedTraceData();
+      });
+
+      after(async () => {
+        await controls.stop();
+      });
+
+      afterEach(async () => {
+        await controls.clearIpcMessages();
+      });
+
+      describe(opts.name, () => {
+        it('metrics from native add-ons should become available at some point', async () => {
+          await retry(() =>
+            Promise.all([
+              //
+              agentControls.getAggregatedMetrics(controls.getPid()),
+              agentControls.getEvents()
+            ]).then(opts.check)
+          );
+        });
+
+        it('should successfully copy the precompiled binaries', async () => {
+          await check(path.join(tmpFolder, 'node_modules', opts.name));
```

Comment:
Thanks again to @aryamohanan for adding this new check. Helped me to verify that the precompiles still work.